pythran/pythonic/types/numpy_expr.hpp

```diff
@@ -74,6 +74,21 @@ namespace pythonic
 
     template <class Op, class... Args>
     template <int... I>
+    bool numpy_expr<Op, Args...>::_no_broadcast(utils::seq<I...>) const
+    {
+      std::initializer_list<bool> nb = {
+          ((size() == std::get<I>(args).shape()[0]) ||
+           (0 == std::get<I>(args).shape()[0]))...};
+      return std::all_of(nb.begin(), nb.end(), [](bool b) { return b; });
```

Comment:
It is details but if you perform your comparisons in the boolean test of all_of instead of doing this in the std::initialiser_list, you "might" compute less comparisons.